### ISSUE TITLE
How to use it in huggingface AdamW optimizer?

### ISSUE THREAD

**Issue by ztl-35** [facebookresearch/bitsandbytes#13]
hi, thanks for this work! I want use it in huggingface AdamW optimizer to train Pre-trained language model, such as BERT. How can I use it, thanks! 

![648C607C-793D-4819-A63D-386E02F43829](https://user-images.githubusercontent.com/31171445/141402444-579406dd-37fa-42cd-8d2b-01622cac9666.png)

**Comment by TimDettmers** [comment]
You should be able to just replace `AdamW` with `bnb.optim.Adam` (which uses the AdamW formulation if your weight decay is > 0.0). Can you try that and report back if there are any issues?

**Comment by ztl-35** [comment]
Hi, I try this method, replacing the AdamW with bnb.optim.Adam. Although output of he task performance is normal, I find the result is much lower than the previous Huggingface AdamW function and I don't change any hyper-paramerters.

**Comment by TimDettmers** [comment]
Can you share the model that you are fine-tuning along with the dataset and results? If you are fine-tuning on GLUE for example there can be quite some noise in the results and the best comparison is done across 5-10 random seeds to get a clear picture of variability between 32-bit and 8-bit AdamW.

**Comment by TimDettmers** [comment]
The problem in this example was that the default `weight_decay` argument is different from `AdamW` and `Adam8bit`. Even though `Adam8bit` implements the same algorithm by default weight decay is turned off. The new `AdamW8bit` class should rectify this since it has the correct default hyperparameter. This was fixed in 2f8083bd8b084290f888fe59b329d98ebd6dd468.
